ui-modules/blueprint-composer/app/components/util/d3-blueprint.js

```diff
@@ -838,11 +827,77 @@ export function D3Blueprint(container, options) {
                 return `M ${sourceNode.x},${sourceY} A ${dr},${dr} 0 0,${sweep} ${m.x},${m.y}`;
             });
 
-        relationData.exit()
+        relationArcs.exit()
             .transition()
             .duration(_configHolder.transition)
             .attr('opacity', 0)
             .remove();
+
+        // ====== RELATIONSHIP LABELS =========
+        // Draw relationship labels that follow paths, somewhere in the middle of paths.
+        // NOTE <textPath/> DECREASES THE UI PERFORMANCE, USE LABELS WITH CAUTION.
+
+        _relationLabels.selectAll('.relation-label').remove(); // Re-draw labels every time, required to refresh changes.
```

Comment:
Re-binding data for multiple labels in a single direction does not resolve the case when paths are re-assigned, especially when there can be more than one relationship type per direction. For that reason - re-draw them every time.